probe-rs/src/debug/debug_info.rs

```diff
@@ -759,13 +792,40 @@ impl DebugInfo {
             };
 
             // Part 1-b: Check LR values to determine if we can continue unwinding.
-            // TODO: ARM has special ranges of LR addresses to indicate fault conditions. We should check those also.
             if let Some(check_return_address) = unwind_registers.get_return_address() {
                 if check_return_address.is_max_value() || check_return_address.is_zero() {
                     // When we encounter the starting (after reset) return address, we've reached the bottom of the stack, so no more unwinding after this.
                     stack_frames.push(return_frame);
                     tracing::trace!("UNWIND: Stack unwind complete - Reached the 'Reset' value of the LR register.");
                     break;
+                } else {
+                    // Part 1-c: If the target current frame is an exception handler, we need to update the `unwind_registers` to match the frame that invoked the exception handler.
+                    if let Some(exception_info) = exception_info {
```

Comment:
Can we move this block into a function please? We have insane rightwards drift here.

Reply:
By the time I included the additional logic in the new function block, to appropriately return `ControlFlow<Error, Option<ExceptionInfo>> `, I ended up with almost the same drift (6 vs. 7 levels), but the code is more complex. Can we please keep it this way?

Reply:
We already have insane rightward drift. It makes everything really tricky to navigate and read. I would like to start putting an end to it :/